### DISCUSSION TITLE
Text stroke scaling

### DISCUSSION THREAD

**Discussion by klavsbuss** [fabricjs/fabric.js#7389] [Q&A] [answered]
Hi.
1. Is there a way I can apply text scaling `scaleX` without affecting `strokeWidth` scale?
2. Why do some letters for [Inter](https://fonts.google.com/specimen/Inter) has this weird outline applied? Only solution i found was using `paintFirst: 'stroke',` but is there solution that includes **transparent fill**?

Help much appreciated! 🙏

![Screenshot 2021-09-22 at 9 33 14 PM](https://user-images.githubusercontent.com/18184893/134401950-8a2be95c-2056-49e2-b71e-fbc63b82c59c.png)

**Comment by melchiar** [comment]
obj.strokeUniform = true

**Reply by klavsbuss** [reply]
Seems that it does not work on text
https://jsfiddle.net/2suy1nva/13/

**Reply by melchiar** [reply]
you're right. would you be able to open a bug report for this? It should be supported for text according to the docs http://fabricjs.com/docs/fabric.Text.html#strokeUniform

**Reply by asturur** [reply]
mmm sadly text stroking does not have the same properties of any other scaling, so no, is not really supported right now.
If we revert the scale factor, the text will go smaller again.
Text is painted in a different way. The only thing you can do if to parse the font with opentype.js and transform it in a path.
Not easy not embedded in fabricJS but would solve your transformation issues.

**Reply by klavsbuss** [reply]
Do you consider `Text to path` parser as feature within fabric or it would be too complex to maintain?
Possible usecases:
Pixel perfect svg output - https://medium.com/@lukasvan3l/fabricjs-opentypejs-for-hallmark-f2878ec019a7
No styling limitations (my usecase)

**Reply by asturur** [reply]
It wouldn't be a fabricJS feature, but more a tutorial on how to get there using openTypeJS.
There could be an officially mantained module that reconsider text measurements in that sense, but you have a series of things to solve, starting from getting access to font files for loading, outside css.

**Comment by klavsbuss** [comment, marked as answer]
`2. Why do some letters for Inter has this weird outline applied?`
Seems that [Google Font](https://fonts.google.com/specimen/Inter?query=inter) font version is outdated. Downloaded latest Inter version from [Github](https://github.com/rsms/inter/) and it fixed `2.` problem. `1.` persist though - reported it as bug.
